File: Frontend/laundry-order-app/src/app/models/order.ts
import { LaundryItem } from './laundry-item';

export enum OrderStatus {
    Pending = 0,
    Approved = 1,
    Rejected = 2,
    Completed = 3
}

export interface Order {
    id: number;
    orderDate: Date;
    laundryItems?: LaundryItem[];
    items?: LaundryItem[];        
    status: OrderStatus | string;
    userId: string;
    userName?: string;
    reason?: string;
    comment?: string;
    customerName?: string;
    customerSurname?: string;
}

export interface CreateOrderRequest {
    orderDate: Date;
    customerName: string;
    customerSurname: string;
    laundryItems: LaundryItem[];
    reason?: string;
    comment?: string;
}

export interface UpdateOrderStatusRequest {
    Status: OrderStatus;
    Reason?: string;
}

File: Backend/LaundryOrderAPI/Models/Order.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;

namespace LaundryOrderAPI.Models
{
    public enum OrderStatus
    {
        Pending,
        Approved,
        Rejected,
        Completed
    }

    public class Order
    {        public Order()
        {
            LaundryItems = new List<LaundryItem>();
            CreatedAt = DateTime.UtcNow;
            Status = OrderStatus.Pending;
        }

        public int Id { get; set; }
        
        [Required]
        public DateTime OrderDate { get; set; }
        
        [Required]
        public string CustomerName { get; set; }
        
        [Required]
        public string CustomerSurname { get; set; }
        
        public string Comment { get; set; }
        
        public string Reason { get; set; }
        
        public DateTime CreatedAt { get; set; }
        
        public DateTime? UpdatedAt { get; set; }
        
        public OrderStatus Status { get; set; }
        
        // Navigation property
        public List<LaundryItem> LaundryItems { get; set; }
        
        // User who created the order
        public string UserId { get; set; }
        public AppUser User { get; set; }
    }
}
